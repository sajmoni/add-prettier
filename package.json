{
  "name": "setup-prettier",
  "version": "0.1.0",
  "description": "Setup prettier with good defaults",
  "keywords": [
    "prettier",
    "cli",
    "prettier-config"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/sajmoni/setup-prettier.git"
  },
  "license": "MIT",
  "author": "sajmoni",
  "type": "module",
  "bin": "dist/index.js",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "rm -rf dist && tsc --project ./tsconfig.prod.json",
    "release": "np --no-tests",
    "test": "npm run build && chmod +x dist/index.js && vitest"
  },
  "prettier": "prettier-config-one",
  "dependencies": {
    "chalk": "5.4.1",
    "execa": "9.5.2",
    "read-pkg": "9.0.1",
    "sort-package-json": "2.14.0",
    "write-pkg": "7.0.0"
  },
  "devDependencies": {
    "@types/node": "22.10.7",
    "get-bin-path": "11.0.0",
    "np": "10.1.0",
    "prettier": "3.4.2",
    "prettier-config-one": "0.2.1",
    "tempy": "3.1.0",
    "tsconfig-one": "0.0.9",
    "typescript": "5.7.3",
    "vitest": "3.0.2"
  },
  "engines": {
    "node": ">=22"
  }
}
