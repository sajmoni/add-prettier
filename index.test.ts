import fs from 'node:fs/promises'
import path from 'node:path'
import { expect, test } from 'vitest'
import { execa } from 'execa'
import { temporaryDirectory } from 'tempy'
import { type PackageJson, readPackage } from 'read-pkg'
import { getBinPath } from 'get-bin-path'

import { name } from './package.json'

test(name, async () => {
  const binPath = await getBinPath()
  if (!binPath) {
    throw new Error('Bin path not found')
  }

  const directory = temporaryDirectory({ prefix: 'hello-world' })
  const unFormatted = 'console.log("hello world");'

  await fs.writeFile(path.join(directory, 'index.js'), unFormatted)

  const { stdout } = await execa(binPath, [], {
    cwd: directory,
    env: {
      FORCE_COLOR: '2',
    },
  })

  const formattedFile = await fs.readFile(
    path.join(directory, 'index.js'),
    'utf-8',
  )
  expect(formattedFile).toEqual("console.log('hello world')\n")

  const packageJson = (await readPackage({
    cwd: directory,
    normalize: false,
  })) as PackageJson & {
    prettier: string
    devDependencies: Record<string, string>
  }

  expect(packageJson.prettier).toEqual('prettier-config-one')
  expect(Object.keys(packageJson.devDependencies)).toEqual([
    'prettier',
    'prettier-config-one',
  ])
  console.log('stdout', stdout)
  expect(stdout).toMatchSnapshot()
})
