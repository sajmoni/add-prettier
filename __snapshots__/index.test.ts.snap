// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`setup-prettier 1`] = `
"
[34m[1m setup-prettier[22m[39m

 [32mprettier and prettier-config-one added![39m
"
`;
